cns/client/client.go

```diff
@@ -403,16 +401,18 @@ func (c *Client) RequestIPs(ctx context.Context, ipconfig cns.IPConfigsRequest)
 	req.Header.Set(headerContentType, contentTypeJSON)
 	res, err := c.client.Do(req)
 
-	// if we get a 404 error
-	if res.StatusCode == http.StatusNotFound {
-		return nil, fmt.Errorf("cannot find API RequestIPs %w: %v", ErrAPINotFound, err) //nolint:errorlint // multiple %w not supported in 1.19
-	}
-
 	if err != nil {
 		return nil, errors.Wrap(err, "http request failed")
 	}
 	defer res.Body.Close()
 
+	if res.StatusCode == http.StatusNotFound {
+		return nil, &CNSClientError{
+			Code: types.UnsupportedAPI,
+			Err:  errors.Errorf("Unsupported API"),
```

Comment:
- this error should be a typed error, not an anonymous error, so that it can be compared to with `errors.Is/As`
- error strings should not be sentence-case ("unsupported API") because they are embedded within other strings